### PR TITLE
ProvVoipEnvia: Fix problems in updating envia TEL contracts.

### PR DESCRIPTION
**Pull request by xee8ai**
#

### CODE REVIEW

File: modules/ProvVoipEnvia/Console/EnviaContractGetterCommand.php [hunk 1]
```diff
@@ -56,65 +62,120 @@ public function handle()
...
      *
      * @author Patrick Reichel
      */
-    protected function _get_contracts()
+    protected function getContracts()
     {
         Log::debug(__METHOD__.' started');
 
-        $contracts = Contract::all();
+        $contracts = Contract::whereNotNull('customer_external_id')->get();
```
Comment (NinoRy) [inline modules/ProvVoipEnvia/Console/EnviaContractGetterCommand.php:96]: This query makes it incredible faster!
```suggestion
        $contracts = Contract::whereNotNull('customer_external_id')->with('modems.mtas.phonenumbers', 'modems.mtas.phonenumbers.phonenumbermanagement')->get();
```

File: modules/ProvVoipEnvia/Console/EnviaContractGetterCommand.php [hunk 2]
```diff
@@ -56,65 +62,120 @@ public function handle()
...
+        $msg = 'Found '.count($contracts).' with external_customer_id set.';
+        Log::debug(__CLASS__.": $msg");
+        $this->line($msg);
 
-        foreach ($contracts as $contract) {
+        $today = date('Y-m-d');
+        $one_year_back = date('Y-m-d', strtotime('-1 year'));
```
[the remark this answers is not in the document]
Comment (olebowle) [reply modules/ProvVoipEnvia/Console/EnviaContractGetterCommand.php]: fixed in d21eb85

File: modules/ProvVoipEnvia/Console/EnviaContractGetterCommand.php [hunk 3]
```diff
@@ -56,65 +62,120 @@ public function handle()
...
             }
-
-            if ($has_active_number) {
-                array_push($this->contracts_to_get_envia_contracts_for, $contract->id);
+            $bar->advance();
+        }
+        $bar->finish();
+
+        // handle enviacontracts for ended nms contracts
+        $enviacontracts = EnviaContract::where('state', 'NOT LIKE', 'Gekündigt')->where('state', 'NOT LIKE', 'Nicht ermittelbar')->get();
```
[the remark this answers is not in the document]
Comment (olebowle) [reply modules/ProvVoipEnvia/Console/EnviaContractGetterCommand.php]: fixed in e3d1968

File: modules/ProvVoipEnvia/Console/EnviaContractGetterCommand.php (hunk 1 above)
Comment (olebowle) [reply modules/ProvVoipEnvia/Console/EnviaContractGetterCommand.php]: fixed in e3d1968

**Pull request merged**